src/renderers/shared/fiber/ReactFiber.js

```diff
@@ -48,12 +58,51 @@ module.exports = function(tag : number) : Fiber {
     input: null,
     output: null,
 
-    handler: function() {},
-    handlerTag: 0,
-
     hasPendingChanges: true,
 
-    stateNode: {},
+    stateNode: null,
 
   };
 };
+
+function shouldConstruct(Component) {
+  return !!(Component.prototype && Component.prototype.isReactComponent);
+}
+
+exports.createFiberFromElement = function(element : ReactElement) {
+  const fiber = exports.createFiberFromElementType(element.type);
+  if (typeof element.type === 'object' && typeof fiber.input === 'function') {
```

Comment:
How can `fiber.input` be set here? It looks like this comes directly from `createFiber` so it would always be null.